src/vshaxe/Main.hx

```diff
@@ -42,6 +42,7 @@ function main(context:ExtensionContext) {
 	final problemMatchers = ["$haxe-absolute", "$haxe", "$haxe-error", "$haxe-trace"];
 	final api = {
 		haxeExecutable: haxeInstallation.haxe,
+		haxelibExecutable: haxeInstallation.haxelib,
```

Comment:
Exposing this directly through a "read-only-interface" turned out to be a mistake for `haxeExecutable`, as some extension (*cough* Kha) abused this and directly manipulates the value. We need to expose the new API in a way that doesn't allow for that, even if you bypass the type system / don't use Haxe to access the API.

Reply:
I see. In that case I think breaking the reference to the actual instance of `HaxeExecutable` and its internal configuration should be enough.